lambdas/data-migration1/tests/test-rules.js

```diff
@@ -0,0 +1,399 @@
+const { dynamodbDocClient } = require('@cumulus/aws-client/services');
+const { fakeCollectionFactory, fakeProviderFactory } = require('@cumulus/api/lib/testUtils');
+const { getKnexClient, localStackConnectionEnv } = require('@cumulus/db');
+const { randomId, randomString } = require('@cumulus/common/test-utils');
+
+// eslint-disable-next-line node/no-unpublished-require
+const { migrationDir } = require('../../db-migration');
+const { migrateCollectionRecord } = require('../dist/lambda/collections');
+const { migrateProviderRecord } = require('../dist/lambda/providers');
+const { migrateRuleRecord, migrateRules } = require('../dist/lambda/rules');
+const { RecordAlreadyMigrated } = require('../dist/lambda/errors');
+
+const testDbName = `data_migration_1_${cryptoRandomString({ length: 10 })}`;
+const testDbUser = 'postgres';
+const workflow = randomId('workflow-');
+const ruleOmitList = ['createdAt', 'updatedAt', 'state', 'provider', 'collection', 'rule'];
+
+const generateFakeRule = (collectionName, collectionVersion, providerId, enabled = true) => ({
+  name: cryptoRandomString({ length: 10 }),
+  workflow: workflow,
+  provider: providerId,
+  state: enabled ? 'ENABLED' : 'DISABLED',
+  collection: {
+    name: collectionName,
+    version: collectionVersion,
+  },
+  rule: { type: 'onetime', value: randomString(), arn: randomString(), logEventArn: randomString() },
+  meta: { key: 'value' },
+  queueUrl: randomString(),
+  payload: { result: { key: 'value' } },
+  tags: ['tag1', 'tag2'],
+  createdAt: Date.now(),
+  updatedAt: Date.now(),
+});
+
+let collectionsModel;
+let providersModel;
+let rulesModel;
+let fakeCollection;
+let fakeProvider;
+
+test.before(async (t) => {
+  process.env.stackName = cryptoRandomString({ length: 10 });
+  process.env.system_bucket = cryptoRandomString({ length: 10 });
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.ProvidersTable = cryptoRandomString({ length: 10 });
+  process.env.RulesTable = cryptoRandomString({ length: 10 });
+
+  const createKeyResponse = await KMS.createKey();
+  process.env.provider_kms_key_id = createKeyResponse.KeyMetadata.KeyId;
+  t.context.providerKmsKeyId = process.env.provider_kms_key_id;
+
+  const workflowfile = `${process.env.stackName}/workflows/${workflow}.json`;
+  const messageTemplateKey = `${process.env.stackName}/workflow_template.json`;
+
+  collectionsModel = new Collection();
+  await collectionsModel.createTable();
+
+  providersModel = new Provider();
+  await providersModel.createTable();
+
+  rulesModel = new Rule();
+  await rulesModel.createTable();
+  await createBucket(process.env.system_bucket);
+
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: false,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  });
```

Comment:
It seems like we need a fakeCollection/fakeProvider for most of the tests, so I'd move this to a `beforeEach` handler for the tests and attach them to `t.context`, like so:

```
test.beforeEach((t) => {
  t.context.fakeCollection = fakeCollectionFactory();
  t.context.fakeProvider = fakeProviderFactory({
    encrypted: false,
    privateKey: 'key',
    cmKeyId: 'key-id',
    certificateUri: 'uri',
    createdAt: Date.now(),
    updatedAt: Date.now(),
  });
});
```

Then, in each test, you'll have a new version of these objects available that you can access like so:

`const { fakeCollection, fakeProvider } = t.context`